packages/common-integration-tests/src/leaderboard-client.ts

```diff
@@ -0,0 +1,1072 @@
+          rank: BigInt(1),
+        },
+        {
+          id: BigInt(123),
+          score: 100,
+          rank: BigInt(2),
+        },
+      ];
+      expect(receivedDescending).toEqual(expectedDescending);
+
+      const topTwo = await leaderboardClient.leaderboardFetchByRank(
+        integrationTestCacheName,
+        leaderboardName,
+        {
+          startRank: 0,
+          endRank: 2,
+          order: LeaderboardOrder.Ascending,
+        }
+      );
+      expectWithMessage(() => {
+        expect(topTwo).toBeInstanceOf(LeaderboardFetch.Found);
+      }, `expected FOUND but got ${topTwo.toString()}`);
+      const receivedTopTwo = (topTwo as LeaderboardFetch.Found).valueArray();
+      const expectedTopTwo = [
+        {
+          id: BigInt(123),
+          score: 100,
+          rank: BigInt(0),
+        },
+        {
+          id: BigInt(456),
+          score: 200,
+          rank: BigInt(1),
+        },
+      ];
+      expect(receivedTopTwo).toEqual(expectedTopTwo);
+    });
+  });
+
+  describe('#Get element rank', () => {
+    const leaderboardName = `leaderboard-${v4()}`;
+
+    it('validates the cache name', async () => {
+      const response = await leaderboardClient.leaderboardGetRank(
+        '   ',
+        leaderboardName,
+        BigInt(123)
+      );
+      expectWithMessage(() => {
+        expect(response).toBeInstanceOf(LeaderboardGetRank.Error);
+      }, `expected ERROR but got ${response.toString()}`);
+      const responseError = response as LeaderboardGetRank.Error;
+      expectWithMessage(() => {
+        expect(responseError.errorCode()).toEqual(
+          MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        );
+      }, `expected INVALID_ARGUMENT_ERROR but got ${responseError.errorCode()} ${responseError.message()}`);
+    });
+
+    it('validates the leaderboard name', async () => {
+      const response = await leaderboardClient.leaderboardGetRank(
+        integrationTestCacheName,
+        '   ',
+        BigInt(123)
+      );
+      expectWithMessage(() => {
+        expect(response).toBeInstanceOf(LeaderboardGetRank.Error);
+      }, `expected ERROR but got ${response.toString()}`);
+      const responseError = response as LeaderboardGetRank.Error;
+      expectWithMessage(() => {
+        expect(responseError.errorCode()).toEqual(
+          MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        );
+      }, `expected INVALID_ARGUMENT_ERROR but got ${responseError.errorCode()} ${responseError.message()}`);
+    });
+
+    // Currently failing with error code 14
+    // it('returns NotFound when leaderboard does not exist', async () => {
+    //   const response = await leaderboardClient.leaderboardGetRank(
+    //     integrationTestCacheName,
+    //     leaderboardName,
+    //     BigInt(123)
+    //   );
+    //   expectWithMessage(() => {
+    //     expect(response).toBeInstanceOf(LeaderboardGetRank.NotFound);
+    //   }, `expected NotFound but got ${response.toString()}`);
+    // });
```

Comment:
Error code 14 is gRPC not found, so isn't that expected? The SDK should transform this to a `NotFound`

Reply:
Oh okay, yeah, I forgot to add that check in the SDK code then. Will fix that now

Reply:
Wait, isn't error code 5 grpc not found? https://grpc.github.io/grpc/core/md_doc_statuscodes.html

Reply:
Oh you're right, sorry, I totally misread my small font on my screen 🙈 Okay if this is getting an error code 14 that means we're doing something wrong, let me confirm some behavior and see what's yielding this bug on the back end

Reply:
Hmmm, local testing passed for me 🤔

Reply:
Ah, I see the issue in our metrics. I can dig deeper into this